reactor-core/src/test/java/reactor/core/publisher/FluxPublishOnTest.java

```diff
@@ -1186,8 +1184,15 @@ public void mapNotifiesOnce() throws InterruptedException {
 			                            }
 		                            }));
 
+		final long start = System.currentTimeMillis();
 		for (int i = 0; i < COUNT; i++) {
-			while (s.emitNext(i).hasFailed() );
+			long busyLoops = 0;
+			while (s.tryEmitNext(i).hasFailed() ) {
+				busyLoops++;
+				if (busyLoops % 5000 == 0 && System.currentTimeMillis() - start >= 10_0000) {
```

Comment:
nit: this whole thing can be replaced with Awaitility, I think

Reply:
I tried, but it makes the test longer and the tuning of poll interval is complicated, I won't have time to fine tune it.